src/producer.rs

```diff
@@ -140,10 +140,13 @@ impl Archive {
     fn check_file(file: FilePath, checker: &Fn(&mut Read) -> bool) -> bool {
         match file {
             FilePath::File(reader) => checker(reader),
-            FilePath::Path(path) => match File::open(path) {
-                Ok(mut f) => checker(&mut f),
-                Err(_) => false,
-            },
+            FilePath::Path(path) => {
```

Comment:
This one in particular was probably a bit more readable with the match instead of if-let